packs/aws/sensors/sqs_sensor.py

```diff
@@ -0,0 +1,95 @@
+"""
+This is generic SQS Sensor using boto3 api to fetch messages from sqs queue.
+After receiving a messge it's content is passed as payload to a trigger 'aws.sqs_new_message'
+
+This sensor can be configured either by using config.yaml withing a pack or by creating
+following values in datastore:
+    - aws.input_queue
+    - aws.aws_access_key_id
+    - aws.aws_secret_access_key
+    - aws.region
+"""
+
+from boto3.session import Session
+from botocore.exceptions import ClientError
+
+from st2reactor.sensor.base import PollingSensor
+
+
+class AWSSQSSensor(PollingSensor):
+    def __init__(self, sensor_service, config=None, poll_interval=5):
+        super(AWSSQSSensor, self).__init__(sensor_service=sensor_service, config=config,
+                                           poll_interval=poll_interval)
+
+    def setup(self):
+        self.input_queue = self._GetConfigEntry(key='input_queue', prefix='sqs_sensor')
+        self.aws_access_key = self._GetConfigEntry('aws_access_key_id')
+        self.aws_secret_key = self._GetConfigEntry('aws_secret_access_key')
+        self.aws_region = self._GetConfigEntry('region')
+
+        self._logger = self._sensor_service.get_logger(name=self.__class__.__name__)
+
+        self.session = None
+        self.sqs_res = None
+
+        self._SetupSqs()
+        self.queue = self._GetQueueByName(self.input_queue)
+
+    def poll(self):
+        msg = self._receive_messages(queue=self.queue)
+        if msg:
+            payload = {"queue": self.input_queue, "body": msg[0].body}
+            self._sensor_service.dispatch(trigger="aws.sqs_new_message", payload=payload)
+            msg[0].delete()
```

Comment:
Alright, so yeah - it behaves like ACK.

In this case the current approach is fine - we only want to ACK if the message has already been processed. If the processing fails (exception is thrown or similar) message won't be ACKed so it will be re-delivered (re-retrieved) and processed again.